lua/plenary/enum.lua

```diff
@@ -0,0 +1,40 @@
+local Enum = {}
+
+function Enum:__eq(other)
+  if getmetatable(other) == nil then return false end
+  for k, v in pairs(self) do
+    if type(k) == 'string' then
+      if v ~= other[k] then return false end
+    else
+      return false
+    end
+  end
+  return true
+end
+
+local function make_enum(tbl)
+  local result = {}
+  local i = 1
+  -- Filter out everything that isn't a string, or a table since we only care
+  -- about those for enum names/values
+  for _, v in ipairs(tbl) do
+    if type(v) == 'string' then
+      result[v] = i
```

Comment:
can we make `i` a metatable maybe called `Variant` which has `__eq` method that can only compare with stuff from the same enum and not with other numbers. Also we can add an `:id` method for each variant to get id